app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,45 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.all
+  end
+
+  def show
+    @market = Market.where(id: params[:id]).first
+    @vendors = Vendor.where(market_id: params[:id])
+  end
+
+  def new
+    @market = Market.new
+    @vendors = Vendor.where(market_id: params[:id])
+  end
+
+  def create
+    @market = Market.new(market_create_params[:market])  #use the params that you accept
+    if @market.save                         #save will return true if it DID save
+      redirect_to markets_path               #redirect them somewhere else if not working
+    else
+      render :new                         #finish the transaction by rendering the view
+    end
+  end
+
+  def edit
+    @market = Market.find(params[:id])
+    render :new
+  end
+
+  def update
+    @market = Market.find(params[:id])
+    @market.update(market_update_params[:market])
+    redirect_to markets_path
+  end
+
+  private
+
+  def market_create_params  #tell rails which params are ok to be in the model
+    params.permit(market: [:name, :address, :city, :county, :state, :zip])
+  end
+
+  def market_update_params  #tell rails which params are ok to be in the model
+    params.permit(market: [:name, :address, :city, :county, :state, :zip])
+  end
```

Comment:
This method and the market_create_params method are identical and could just be one method that you reuse.